src/main/java/amazin/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,59 @@
+package amazin.controller;
+
+import amazin.model.Book;
+import amazin.model.Item;
+import amazin.model.ShoppingCart;
+import amazin.repository.BookRepository;
+import amazin.service.ShoppingCartService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
+import java.lang.reflect.Array;
+import java.util.*;
+
+/**
+ * Created by lauramachado on 2019-03-18.
+ */
+@Controller
+@SessionAttributes("cart")
+public class ShoppingCartController {
+
+    @Autowired
+    ShoppingCartService shoppingCartService;
+
+    @Autowired
+    BookRepository bookRepository;
+
+    public static final String VIEW_SHOPPING_CART = "shopping-cart";
+
+    @GetMapping("/cart/delete/{bookId}")
+    public String deleteCartItem(@PathVariable("bookId") Long bookId, HttpSession session){
```

Comment:
delete action should not be mapped for a Get mapping Get requests are for read only operations

Reply:
this should be a Post instead

Reply:
why's addToCart() in ShoppingCartRestController but deleteCartItem is here in ShoppingCartController?